confdb/confdb_test.go

```diff
@@ -3855,3 +3858,155 @@ func (*viewSuite) TestFieldFilterPathMismatch(c *C) {
+			// or it's an index and we're expecting them
+			(confdb.ValidIndexSubkey.MatchString(ss) && !o.ForbidIndexes) ||
+			// or it's an index placeholder and we're expecting them
+			(confdb.ValidIndexPlaceholder.MatchString(ss) && o.AllowPlaceholders) {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
+func hasEmptySubkey(s string, o confdb.ParseOptions) bool {
+	subStrings := subkeyWithDot.FindAllString(s, -1)
+	if len(subStrings) == 0 {
+		return true
+	}
+	if subStrings[0] == "." || subStrings[len(subStrings)-1] == "." {
+		return true
+	}
+	// Only if partial paths are allowed can a path begin with '['
+	if strings.HasPrefix(subStrings[0], "[") {
+		return true
+	}
+	for i, ss := range subStrings {
+		// A subkey can never begin with a '[' unless it's the first rune
+		// Two successive '.' means an empty string subkey
+		if i > 0 && subStrings[i-1] == "." && (strings.HasPrefix(ss, "[") || ss == ".") {
+			return true
+		}
+	}
+	return false
+}
+
+func hasUnexpectedFieldFilterTermination(s string) bool {
+	subStrings := subkeyWithDot.FindAllString(s, -1)
+	for i, ss := range subStrings {
+		// The regex will break up any sequence that has "[." not followed by a closing ']'
+		// The '[' with therefore always be alone as well as the '.'
+		if i > 0 && subStrings[i-1] == "[" && ss == "." {
+			return true
+		}
+	}
+	return false
+}
+
+func isProblematicSubkeyError(s string) bool {
+	return strings.Contains(s, "invalid subkey") ||
+		strings.Contains(s, "must conform to base format") ||
+		strings.Contains(s, "field filters, if present, must be at the end of a sub-key")
+}
+
+func isFieldFilterError(s string) bool {
+	return strings.Contains(s, "field filter terminated unexpectedly") ||
+		strings.Contains(s, "field filter must be in the format") ||
+		strings.Contains(s, "invalid field filter")
+}
+
+func fuzzHelper(f *testing.F, o confdb.ParseOptions, seed string) {
+	f.Add(seed)
+	f.Fuzz(func(t *testing.T, s string) {
+		accessors, err := confdb.ParsePathIntoAccessors(s, o)
+		if err != nil && isProblematicSubkeyError(err.Error()) && hasInvalidSubkeys(s, o) {
+			t.Skip()
+		}
+		if err != nil && err.Error() == "cannot have empty subkeys" && hasEmptySubkey(s, o) {
+			t.Skip()
+		}
+		if err != nil && err.Error() == "non UTF-8 character" && !utf8.ValidString(s) {
+			t.Skip()
+		}
+		if err != nil && isFieldFilterError(err.Error()) && hasUnexpectedFieldFilterTermination(s) {
+			t.Skip()
+		}
+		if err != nil && strings.Contains(err.Error(), "cannot apply more than one filter to the same field") {
+			t.Skip()
+		}
+		if err != nil {
+			t.Errorf("encountered error %s with input %s", err, s)
+		}
+		expected := subkeyOnlyReg.FindAllString(s, -1)
+		if len(accessors) == len(expected) {
+			for i, e := range expected {
+				if accessors[i].Access() != e {
+					t.Errorf("unexpected type of accessor %v with name %s for element %s", accessors[i].Type(), accessors[i].Name(), e)
```

Comment:
tiny nitpick: I think the type will printed as an int? It might be nicer to use `%T` which prints the accessors type (mapKey, keyPlaceholder, etc).